File: samsara/core/src/main/scala/samsara/MainMenuScreen.scala
//
// Mashups - a series of mashed up game prototypes
// https://github.com/samskivert/mashups/blob/master/LICENSE

package samsara

import tripleplay.game.ScreenStack
import tripleplay.ui._
import tripleplay.ui.layout.AxisLayout
import tripleplay.ui.layout.TableLayout

class MainMenuScreen (game :Samsara) extends ScreenStack.UIScreen(game.plat) {

  def game = game

  override def wasAdded () {
    val root = iface.createRoot(AxisLayout.vertical, UI.sheet(game.plat), layer)
    root.addStyles(Style.BACKGROUND.is(Background.solid(0xFFFFFFFF)))

    val flickTapBox = new CheckBox()
    flickTapBox.select(game.flickTapInput)
    flickTapBox.selected.connect(slot[java.lang.Boolean] { case checked =>
      game.flickTapInput = checked
    })
    val flickBox = new CheckBox()
    flickBox.select(game.flickInput)
    flickBox.selected.connect(slot[java.lang.Boolean] { case checked =>
      game.flickInput = checked
    })
    val relTapBox = new CheckBox();
    relTapBox.select(game.absTapInput)
    relTapBox.selected.connect(slot[java.lang.Boolean] { case checked =>
      game.relTapInput = checked
    })
    val absTapBox = new CheckBox();
    absTapBox.select(game.absTapInput)
    absTapBox.selected.connect(slot[java.lang.Boolean] { case checked =>
      game.absTapInput = checked
    })

    root.add(UI.stretchShim,
             new Label("Sa\u1E43s\u0101ra").addStyles(Style.FONT.is(UI.titleFont)),
             UI.stretchShim,
             new Group(new TableLayout(TableLayout.COL, TableLayout.COL.alignLeft()).
      gaps(5, 5)).add(
      flickTapBox, new Label("Flick+tap input"),
      flickBox, new Label("Flick input"),
      relTapBox, new Label("Relative tap input"),
      absTapBox, new Label("Absolute tap input")),
             UI.stretchShim,
             new Button("New Game").addStyles(Style.FONT.is(UI.menuFont)).onClick(newGame),
             UI.stretchShim)
    root.setSize(size)
  }

  protected def newGame () {
    game.screens.push(new GameScreen(game, new LevelDB), game.screens.slide.down)
  }
}

File: ziggurat/core/src/main/java/ziggurat/core/MainMenuScreen.java
//
// Mashups - a series of mashed up game prototypes
// https://github.com/samskivert/mashups/blob/master/LICENSE

package ziggurat.core;

import rsp.State;
import tripleplay.ui.Label;
import tripleplay.ui.Root;
import tripleplay.ui.layout.AxisLayout;

public class MainMenuScreen extends Screen {

  public static class S extends ScreenS<Ziggurat> {
    public Screen create (Ziggurat ctx) {
      return new MainMenuScreen(ctx);
    }
  }

  public MainMenuScreen (Ziggurat ctx) {
    super(ctx.root, "main");
  }

  // @Override public void wasShown () {
  //   super.wasShown();
  //   _root = iface.createRoot(AxisLayout.vertical(), UI.stylesheet(), layer);
  //   // _root.addStyles(Style.BACKGROUND.is(background()));
  //   _root.setSize(width(), height());
  //   // create your user interface
  //   _root.add(new Label("Ziggurat"));
  // }

  // // destroy your user interface in wasHidden (or wasRemoved)
  // @Override public void wasHidden () {
  //   super.wasHidden();
  //   iface.destroyRoot(_root);
  // }

  // protected Root _root;
}
